webapp/src/contexts/ConferenceContext/methods/toggleMuteVideo.ts

```diff
@@ -1,8 +1,35 @@
 import { ConferenceActionType, type ConferenceAction } from '../ConferenceAction'
 import { type ConferenceState } from '../ConferenceState'
 
-export const toggleMuteVideo = async (state: ConferenceState, dispatch: React.Dispatch<ConferenceAction>): Promise<void> => {
-  const { client, videoMuted } = state
+export const toggleMuteVideo = async (
+  state: ConferenceState,
+  dispatch: React.Dispatch<ConferenceAction>
+): Promise<void> => {
+  const { client, videoMuted, localStream } = state
+
+  if (!videoMuted) {
+    localStream?.getVideoTracks().forEach((track) => {
+      track.stop()
+    })
+  } else {
+    const newStream = await navigator.mediaDevices.getUserMedia({
+      audio: false,
+      video: { deviceId: state.inputVideoDeviceId }
+    })
+
+    localStream?.getAudioTracks().forEach((track) => {
```

Comment:
Why do we need to  merge those tracks if they are decoupled? Should we just do it only when we set them in Peer-Connection obj? and btw then you shouldnt have the problem with https://github.com/pexip/pexip-mattermost-plugin/pull/23#discussion_r1609626155 as well, as we dont really need this track locally only when we set it to the remote

Reply:
The thing is that I want to have the `AudioTracks` also locally, because if I click on `muteAudio`, I stop the `AudioTracks`. When clicked again, I call to `getUserMedia()` to ask only for the `AudioTracks`.